packages/sd-jwt/src/verify.ts

```diff
@@ -0,0 +1,119 @@
+import * as jwt from "jsonwebtoken";
+import * as jwks from "jwks-rsa";
+import { SdJwt, SupportedAlgorithm } from "./types";
+
+type VerifyOptions = { jwksUri: string };
+type VerifyResult = SdJwt;
+
+const supportedAlgorithm = new Set<SupportedAlgorithm>([
+  "RS256",
+  "RS384",
+  "RS512",
+  "ES256",
+  "ES384",
+  "ES512",
+  "PS256",
+  "PS384",
+  "PS512",
+]);
+function isSupportedAlgorithm(alg: string): alg is SupportedAlgorithm {
+  return supportedAlgorithm.has(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    alg as any /* to look for unknown value into a typed set */
+  );
+}
+
+// Check a jwt has the attributes defined in the SD-JWT specification
+// It only support first level of disclosure for now
+function jwtIsSdJwt(j: jwt.Jwt): j is SdJwt {
+  // every condition expect this, we can short-circuit the check
+  if (typeof j.payload !== "object") {
+    return false;
+  }
+
+  const hasValidSd =
+    "_sd" in j.payload &&
+    Array.isArray(j.payload["_sd"]) && // eslint-disable-line @typescript-eslint/dot-notation
+    j.payload["_sd"].every((s) => typeof s === "string"); // eslint-disable-line @typescript-eslint/dot-notation
+  const hasValidSdAlg =
+    "_sd_alg" in j.payload && typeof j.payload["_sd_alg"] === "string"; // eslint-disable-line @typescript-eslint/dot-notation
+  const hasNotSdAlg = !("_sd_alg" in j.payload);
+
+  return hasValidSd && (hasValidSdAlg || hasNotSdAlg);
```

Comment:
What do you think about this little refactor?

```suggestion
  const payload = j.payload as SdJwt["payload"];

  const hasValidSd =
    "_sd" in payload &&
    Array.isArray(payload["_sd"]) && // eslint-disable-line @typescript-eslint/dot-notation
    payload["_sd"].every((s) => typeof s === "string"); // eslint-disable-line @typescript-eslint/dot-notation
  const hasValidSdAlg =
    "_sd_alg" in payload && typeof payload["_sd_alg"] === "string"; // eslint-disable-line @typescript-eslint/dot-notation
  const hasNotSdAlg = !("_sd_alg" in payload);

  return hasValidSd && (hasValidSdAlg || hasNotSdAlg);
```